lib/config/folderconfiguration.go

```diff
@@ -39,12 +37,12 @@ type FolderConfiguration struct {
 	MaxConflicts          int                         `xml:"maxConflicts" json:"maxConflicts"`
 	DisableSparseFiles    bool                        `xml:"disableSparseFiles" json:"disableSparseFiles"`
 	DisableTempIndexes    bool                        `xml:"disableTempIndexes" json:"disableTempIndexes"`
-	Fsync                 bool                        `xml:"fsync" json:"fsync"`
 	Paused                bool                        `xml:"paused" json:"paused"`
 	WeakHashThresholdPct  int                         `xml:"weakHashThresholdPct" json:"weakHashThresholdPct"` // Use weak hash if more than X percent of the file has changed. Set to -1 to always use weak hash.
 
-	cachedPath string
+	cachedFilesystem fs.Filesystem
 
+	DeprecatedPath           string  `xml:"path,attr,omitempty" json:"-"`
```

Comment:
Compatibility: we talked about this, and I can indeed fix it up later if you prefer, but I would aim for `json:"path,omitempty"` and keep that attribute populated as long as the type is `basic`. On input, if `DeprecatedPath` is set and the `Filesystem*` are empty, set the filesystem stuff appropriately.